crawler/obfuscate.rb

```diff
@@ -0,0 +1,16 @@
+module Obfuscate
+    def self.email ( h )
+        if h.is_a? Hash
+            if h['email'] && !h['email'].empty?
+                email = h.delete('email')
+                name, host = email.split("@")
+                h['email'] = [Digest::SHA1.hexdigest(name), host].compact.join("@")
+            end
+
+            h.each_value do |v|
+                self.email(v) if v.is_a? Hash
+                v.each {|e| self.email(e)} if v.is_a? Array
```

Comment:
There is an unhandled edge case here that recently took down the crawler. I believe the issue was that we had an array of strings and that crashed this function.

Reply:
Yeah I believe this is captured in the test - see test_no_obfuscation_for_other_fields which has an "array-string" value.

This bubbled up in local testing and was before I found the fix in prod.